### PR TITLE
Print the ValidationResults in various formats

### PR DESCRIPTION
**Pull request by kdallmeyer-sr**
Add ability to print the ValidationResults to various formats (Console, Log, HTML, CSV, Text) as well as watch the validation summary during the run.

```kotlin
val validation: Validation = dataDesc.validate(data) // Not yet executed
val watchedResults = validation.results.watch() // Not yet executed
HtmlPrinter("validation.html").print(validationSummary.results) // Executed and printed!
```

Relates to #216

### CODE REVIEW

File: baleen/src/main/kotlin/com/shoprunner/baleen/DataDescription.kt [hunk 1]
```diff
@@ -53,19 +53,23 @@ class DataDescription(
...
+            var hasError = false
+            allTests.flatMap { it(ctx.dataTrace, ctx.data) }.forEach {
+                if (it is ValidationError) {
+                    hasError = true
+                }
+                yield(it)
+            }
+            if (!hasError) {
+                yield(ValidationSuccess(ctx.dataTrace, ctx.data))
+            }
```
Comment (kdallmeyer-sr) [inline baleen/src/main/kotlin/com/shoprunner/baleen/DataDescription.kt:70]: This makes the evaluation lazy. Before it was outputing to a list and storing results in memory.

**Review by HawaiianSpork — COMMENTED** [review]
This is neat but could you move this to another package.  I have been working on a branch to support multiple platforms and would prefer not have to figure out how to deal with `java.io.File`.  Alternatively if the interface provided a stream instead of a File output that should work.

**Comment by HawaiianSpork** [comment]
I think I would recommend the following new projects:
- `baleen-printer-csv`
- `baleen-printer-html`
- `baleen-printer-text`
- `baleen-printer-java-logger`

They each have their own dependencies so would be good to not link them all together.

You may also want to consider using `OutputStreamWriter` instead of `File`.  It looks like the `append` method opens and closes the file every call.

**Comment by kdallmeyer-sr** [comment]
@HawaiianSpork updated PR with separate modules and OutputStreamWriter.

**Review by HawaiianSpork — APPROVED** [review]
LGTM

**Pull request merged**